app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailViewModel.kt

```diff
@@ -63,6 +62,10 @@ class TaskDetailViewModel @Inject constructor(
     private val _isTaskDeleted = MutableStateFlow(false)
     private val _taskAsync = tasksRepository.getTaskStream(taskId)
         .map { handleResult(it) }
+        .catch {
+            showSnackbarMessage(R.string.loading_task_error) // TODO: Move to separate function?
```

Comment:
Do we need to show this snackbar message? The function that handles the async response should be the one acting when that happens

Reply:
In a real app you might be on the task detail screen when a network sync fails. The message could indicate that the current task might not be up to date (bit tenuous but I was looking for places to show exception handling).

Have refactored to emit Async.Error`.